enhancements/microshift/lvms-topolvm-downstream-replacement.md

```diff
@@ -0,0 +1,269 @@
+we believe that we can provide a sufficient alternative to the TopoLVM 
+upstream with LVMS while maintaining full backwards compatibility 
+within MicroShift.
+
+We want to create a RFE that encompasses this change and allows 
+the MicroShift team to review / acknowledge / veto the proposal 
+based on our findings and suggestions.
+
+### User Stories
+
+As an edge device administrator, I want to make the most of my resources
+and do not want to use various CSI sidecars that introduce unnecessary
+memory and CPU overhead. I want to use a single, lightweight solution
+that is easy to maintain and provides the same functionality as the
+current solution.
+
+As a MicroShift user using the TopoLVM storage driver, I want to ensure
+that my previously created and used storage classes, deviceClasses and PVCs
+are still usable & upgradeable after the change to the downstream driver.
+
+As a MicroShift PM, I want to ensure that LVMS means the same feature set
+to the end user in MicroShift as it does in OpenShift.
+
+As a LVMS maintainer, I do not want to keep maintaining the bundled TopoLVM
+images only for MicroShift. I want it to switch so that maintenance and 
+support can be done solely based on the LVMS project.
+
+As a LVMS maintainer, I want to ensure that the LVMS project is used
+in as many places as possible to ensure that it is well tested and
+maintained.
+
+### Goals
+
+* Switch from using repackaged registry.redhat.io/lvms4/topolvm-rhel9 images
+  to using a minified version of LVMS in MicroShift based on registry.redhat.io/lvms4/lvms4-lvms-operator-bundle.
+* Switch from using CSI Sidecars to using the LVMS Deployment optimized for Edge.
+
+### Non-Goals
+
+* Ensuring that LVMCluster works for all use cases in MicroShift.
+  Instead, we will focus on allowing custom deviceClasses to be used 
+  outside the LVMCluster Custom Resource.
+* Integrating all usual MicroShift TopoLVM use cases into LVMS. (think of LVM RAID, etc.)
+  Instead, we will rebrand our documentation and put forward best practices 
+  for using LVMS in MicroShift based on LVMCluster with an extended documentation
+  for specialised use cases.
+* Forcing MicroShift users to migrate to LVMS CustomResources. We will allow
+  the use of the old deviceClass mechanism without a migration path and permanently
+  allow custom deviceClass management.
+
+## Proposal
+
+With 4.16 and the [controller consolitation initative in LVMS](https://issues.redhat.com/browse/OCPEDGE-689)
+we no longer use any of our own upstream images like TopoLVM or the CSI Sidecars. 
+We would like to introduce a conformant way to rebase MicroShift based on LVMS, not TopoLVM in the future.
+This will allow us to no longer ship topolvm as a component,
+but embed it into LVMS fully as well as maintain LVMS as the only
+storage driver in MicroShift and OpenShift we have to support for lvm2 support.
+
+### Workflow Description
+
+1. Edge device administrator deploys a host with MicroShift 4.15
```

Comment:
Since the target version is 4.17 will we support upgrade from both versions or just from the latest?
Please, note that MicroShift will support Y-2 upgrades starting from 4.16.
```suggestion
1. Edge device administrator deploys a host with MicroShift 4.15 or 4.16
```